internal/config/config.go

```diff
@@ -87,18 +87,24 @@ var reEnv = regexp.MustCompile(`\$\{[a-zA-Z_][a-zA-Z_0-9]*\}`)
 // Parse returns a parsed Lava configuration given a path to a
 // file.
 func Parse(path string) (Config, error) {
-	dag, err := NewConfigGraph(path)
+	graph, err := NewConfigGraph(path)
 	if err != nil {
 		return Config{}, fmt.Errorf("build dag: %w", err)
 	}
-	if err = dag.Resolve(); err != nil {
+	if err = graph.Resolve(); err != nil {
 		return Config{}, fmt.Errorf("resolve dag: %w", err)
 	}
 
-	if err = dag.configs[path].validate(); err != nil {
+	cfg, err := graph.Config(path)
+	if err != nil {
+		return Config{}, fmt.Errorf("unknown path: %v: %w", path, err)
```

Comment:
Given that the `ConfigGragh.Config` already returns the passed URL in the error, I'd simplify this one:

```suggestion
		return Config{}, fmt.Errorf("unknown configuration: %w", err)
```